### DISCUSSION TITLE
How to properly use cypress select()

### DISCUSSION THREAD

**Discussion by gabrielbahniuk** [cypress-io/cypress#16634] [Questions and Help] [answered]
Hi everyone,

I am struggling with the `select()` function from cypress (version ^7.3.0) even after reading the docs.

The select element:
```
<select id="riskLevelSelector">
  <option value="3">3</option>
  <option value="4">4</option>
  <option value="5">5</option>
  <option value="6">6</option>
  <option value="7">7</option>
  <option value="8">8</option>
  <option value="9">9</option>
  <option value="10">10</option>
  <option value="11">11</option>
  <option value="12">12</option>
  <option value="13">13</option>
  <option value="14">14</option>
  <option value="15">15</option>
  <option value="16">16</option>
  <option value="17">17</option>
  <option value="18">18</option>
  <option value="19">19</option>
  <option value="20">20</option>
  <option value="21">21</option>
  <option value="22">22</option>
  <option value="23">23</option>
  <option value="24">24</option>
  <option value="25">25</option>
</select>
```

On Cypress I test it like this (please note the default value from the select is option 3).

`cy.get("select[id='riskLevelSelector']").select('4').should('have.value', '4')`

and when running `cypress open` it returns ok to the assertion and it changes actually the value, because there is a ProgressBar component which goes to the right by increasing the values of the select above. And in fact, it goes to the right because the test changes it from 3 to 4.

The problem is: the frame still displays 3 in the "before" and "after".

Am I doing something wrong?

Thanks in advance!

**Comment by jennifer-shehane** [comment, marked as answer]
If you're referring to when you hover over each command in the Command Log, the select value is not restored during this 'time travel' and will still reflect the default value. Issue is here: https://github.com/cypress-io/cypress/issues/4443 Is this what you mean?

**Reply by gabrielbahniuk** [reply]
Maybe #4443 reports a similar problem, I am not sure if is the same.

I attached a screenshot to better explain it:

![ksnip_20210525-010554](https://user-images.githubusercontent.com/7410668/119417290-c864f100-bcf5-11eb-8e24-4c896cf91fcb.png)

Even though the value seems to be updated, in the middle of the screen it still shows old value.

Is that the same issue?

**Reply by jennifer-shehane** [reply]
Yes, this is https://github.com/cypress-io/cypress/issues/4443.

**Reply by gabrielbahniuk** [reply]
@jennifer-shehane ok, thank you. I will follow up https://github.com/cypress-io/cypress/issues/4443 then.
